btas/generic/converge_class.h

```diff
@@ -132,6 +132,8 @@ namespace btas {
           fitOld_ = -1.0;
           return true;
         }
+      } else if(fit > 0.999){
```

Comment:
why is this hardwired?

Reply:
Should we just check the norm of the gradient then? The change in factors is actually not a very good objective function ... due to the lack of guarantees of uniqueness of CP factors can change significantly without affecting the fit quality much ... gradient norm should be better convergence criterion?

Reply:
I changed fitOld_ to start at 1 instead of -1 to reduce iterations when fit is ~1

Reply:
This does not check the change in the factor matrices, it looks at the change in the 2-norm fit `fit_i = 1.0 - |T - \hat{T}_i| / |T| `. and stops when `fit_i - fit_{i -1} < epsilon`